src/app/services/ws.service.ts

```diff
@@ -0,0 +1,33 @@
+import { SocketIOController, WebsocketContext, wsController } from '@foal/socket.io';
+import { ChatController } from '../controllers/api/chat.controller'
+import { Config } from '@foal/core';
+
+const prefix = Config.get('api_prefix', 'string', '');
+
+export class WebsocketService extends SocketIOController {
+    constructor() {
+        super();
+        this.options = {
+            cors: {
+                origin: '*', 
+            },
+            path: `${prefix}/api/hci-socket`
+        }
+
+    } 
+    subControllers = [
+        wsController('/chat', ChatController)
+    ];
+
+    async onConnection(ctx: WebsocketContext) {
+        const tenantID = ctx.socket.handshake.query.tenantID as string; 
+        const tenantRoom = `tenant_${tenantID}Room`;
+        ctx.socket['tenantID'] = tenantID;
+        ctx.socket.join(tenantRoom);
+
+        ctx.socket.on('disconnect', function() {
+            console.log(`${new Date().toISOString()} onDisconnection`)
```

Comment:
Is this actually needed, or is it just going to create log clutter?

Reply:
We can live without but
` const tenantID = ctx.socket.handshake.query.tenantID as string; 
   const tenantRoom = `tenant_${tenantID}Room`;
   ctx.socket['tenantID'] = tenantID; ` 
   are going to be needed

Reply:
I'm specifically referring to the console log here

Reply:
ohh ill get rid of that there no need for it